src/pam/pam_oslogin_admin.cc

```diff
@@ -86,11 +86,21 @@ PAM_EXTERN int pam_sm_acct_mgmt(pam_handle_t *pamh, int flags, int argc,
                  user_name);
       std::ofstream sudoers_file;
       sudoers_file.open(filename.c_str());
-      sudoers_file << user_name << " ALL=(ALL) NOPASSWD: ALL"
-                   << "\n";
-      sudoers_file.close();
-      chown(filename.c_str(), 0, 0);
-      chmod(filename.c_str(), S_IRUSR | S_IRGRP);
+      // OS Login directories are created by another product, guest-agent
+      // https://github.com/GoogleCloudPlatform/guest-agent/blob/56988fa888b46dc0796a958929dceed460f7a3e8/google_guest_agent/oslogin.go#L344
+      // We should be sure a file is opened for writing
+      if (sudoers_file.is_open()) {
+          sudoers_file << user_name << " ALL=(ALL) NOPASSWD: ALL\n";
+          sudoers_file.close();
+
+          chown(filename.c_str(), 0, 0);
+          chmod(filename.c_str(), S_IRUSR | S_IRGRP);
+      } else {
+          PAM_SYSLOG(pamh, LOG_INFO,
```

Comment:
if a file is not writable, then a user will still be logged in, but without an ability to use sudo (I tested this).
If we fail, then a user won't be logged in, it will be rejected, with no understanding of what is going on. The problem is that the user will see it this way : a new VM is created (with OS Login enabled) but no way to log in. 
If we do not fail, a user get allowed to login and get only partial functionality (no sudo permission)